src/linear_algebra/matrix.lean

```diff
@@ -996,6 +1027,79 @@ det_reindex_self' e A
 
 end reindexing
 
+section lmul
+
+variables {R S T : Type*} [comm_ring R] [comm_ring S] [comm_ring T]
+variables [algebra R S] [algebra S T] [algebra R T] [is_scalar_tower R S T]
+variables {m n : Type*} [fintype m] [decidable_eq m] [fintype n] [decidable_eq n]
+variables {b : m → S} (hb : is_basis R b) {c : n → T} (hc : is_basis S c)
+
+open algebra
+
+@[simp] lemma lmul_algebra_map (x : R) :
+  lmul R S (algebra_map R S x) = algebra.lsmul R S x :=
+linear_map.ext (λ s, by simp [smul_def''])
+
+lemma to_matrix_lmul' (x : S) (i j) :
+  linear_map.to_matrix hb hb (lmul R S x) i j = hb.repr (x * b j) i :=
+by rw [linear_map.to_matrix_apply', lmul_apply]
+
+@[simp] lemma to_matrix_lsmul (x : R) (i j) :
+  linear_map.to_matrix hb hb (algebra.lsmul R S x) i j = if i = j then x else 0 :=
+by { rw [linear_map.to_matrix_apply', algebra.lsmul_coe, linear_map.map_smul, finsupp.smul_apply,
+         hb.repr_self_apply, smul_eq_mul, mul_boole],
+     congr' 1; simp only [eq_comm] }
+
+/-- `matrix.lmul hb x` is the matrix corresponding to the linear map `λ y, x * y`.
```

Comment:
Does the `l` mean` left` or `linear`? My understanding was that it referred to `linear` in `algebra.lmul`, but I may be mistaken.

Reply:
I thought it was `left`, but I guess `lsmul` wouldn't make sense. Still, I would like to keep the name `lmul`, since it matches `algebra.lmul` (and `matrix.mul` already exists).

Reply:
How about `algebra.left_mul_matrix`, described as something like
```
`matrix.lmul hb x` is the matrix that acts on the basis `b` of S as left multiplication by `x` acts on S
```